qstack/spahm/rho/utils.py

```diff
@@ -69,34 +73,101 @@ def dm_open_mod(dm, omod):
 def get_xyzlist(xyzlistfile):
     return np.loadtxt(xyzlistfile, dtype=str, ndmin=1)
 
+def check_data_struct(fin, local=False):
+    x = np.load(fin, allow_pickle=True)
+    if type(x.flatten()[0]) == str or type(x.flatten()[0]) == np.str_:
+        is_labeled = True
+        if not local and x.shape[0] == 1:
+            is_single = True
+        elif  x.shape[1] == 2:
+            is_single = True
+        else:
+            is_single = False
+    else:
+        is_labeled = False
+        if not local and x.ndim == 1:
+            is_single = True
+        elif x.shape[1] != 2: ## could be problematic! (if it's a set of local representations and nfeatures = 2 !!
```

Comment:
although a bit uncommon, I'm trying to find a way around it

Reply:
What type on info would you put in `storage_type` if not None ?

Reply:
string that you parse or maybe better a tuple `(is_labeled, is_single, local)` or smth

Reply:
I get your idea!
But then it would be part of the `load_reps()` kwarg and I thought part of the point was to avoid ambiguity there haha

But I can add it, it might help people to get it right!

Reply:
if it's None by default (hence auto format determination) then it's for "experienced users" so should be fine